src/views/BillDisplay/BillDisplay.jsx

```diff
@@ -3,135 +3,444 @@ import "./styles.scss";
+            },
+            {   title: "Status",
+                onClick: () => this.setState({filter: {
+                                                        ...this.state.filter,
+                                                        statusOpened: true,
+                                                        categoryOpened:false,
+                                                        dateOpened: false
+                                                    },
+                                                    dateFilters: {...this.state.dateFilters, startDate: {...this.state.dateFilters.startDate, selected: false}, endDate: {...this.state.dateFilters.endDate,selected: false} }})
+            },
+            {   title: "A to Z",
+                onClick: () => this.updateBills("A to Z")
+            },
+            {   title: "Z to A",
+                onClick: () => this.updateBills("Z to A")
+            },
+            {   title: "Category",
+                onClick: () => this.setState({filter: {
+                                                        ...this.state.filter,
+                                                        categoryOpened: true,
+                                                        statusOpened: false,
+                                                        dateOpened: false
+                                                    },
+                                                dateFilters: {...this.state.dateFilters, startDate: {...this.state.dateFilters.startDate, selected: false}, endDate: {...this.state.dateFilters.endDate,selected: false} }})
+            },
+            {   title: "Date",
+                onClick: () => this.setState({filter: {
+                                                        ...this.state.filter,
+                                                        dateOpened: true,
+                                                        categoryOpened: false,
+                                                        statusOpened: false
+                                                    }})
+            }
+        ];
+
+     return( 
+      <div className="bill__filter__container">
+        <div className="filter__header" onClick={() =>
+              this.setState({
+                filter: {
+                  ...this.state.filter,
+                  opened: false,
+                }
+              })
+            }> <FaBars size={24}/><span className="filter__header__title">Filters</span> 
         </div>
-    
+        <div className="filter__selections__container">
+            {
+                filterBadges.map((badgeObject, key) =>  <Badge key={key} outline pill onClick={badgeObject.onClick}>{badgeObject.title}</Badge> )
+            }
+        </div>
+        {  (this.state.filter.dateOpened  || this.state.filter.statusOpened === true || this.state.filter.categoryOpened === true) &&
+        <div className="advanced__filter__selection">
```

Comment:
WIth just a quick glance, you can tell this code is really hectic to read. Why don't you separate this into another component

Reply:
Ill leave the naming up to you